cli/src/table.rs

```diff
@@ -0,0 +1,49 @@
+use std::fmt::Display;
+
+const PADDING: usize = 3;
+pub struct Column<T> {
+    header: &'static str,
+    value: Box<dyn Fn(&T) -> String>,
+}
+
+pub struct Table<'a, T> {
+    pub cols: Vec<Column<T>>,
+    pub data: &'a [T],
+}
+
+impl<T> Column<T> {
+    pub fn new(header: &'static str, value: Box<dyn Fn(&T) -> String>) -> Column<T> {
+        Column { header, value }
+    }
+
+    fn width(&self, rows: &[T]) -> usize {
+        match rows.iter().map(|t| (self.value)(t).len()).max() {
+            Some(width) => self.header.len().max(width) + PADDING,
+            None => self.header.len() + PADDING,
+        }
+    }
+}
+
+impl<'a, T> Display for Table<'a, T> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let column_widths = self
+            .cols
+            .iter()
+            .map(|col| col.width(self.data))
+            .collect::<Vec<usize>>();
+        // Print headers
+        for (col, width) in self.cols.iter().zip(&column_widths) {
+            write!(f, "{:width$}", col.header, width = width)?;
+        }
+        writeln!(f)?;
```

Comment:
nit: in general, it's more idiomatic to avoid using the formatting macros when there isn't actually any interpolation; clippy would suggest that we just write
```suggestion
        f.write_char('\n')?;
```
to emit only a newline. not a blocker though!